File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Events/OnHurtBeforeMessage.ts
import { IMessage } from "../../../../Core/ObserverMessageKit";
import DamageInfo from "../DataDefine/DamageInfo";
import AbstractCreature from "../Unit/AbstractCreature";

export class OnHurtBeforeMessage implements IMessage{
    public static readonly Id = "OnHurtBeforeMessage";
    public constructor(public Damage:DamageInfo,public target:AbstractCreature){

    }
}

File: Unity/Assets/Scripts/ModTheSpire/Event/Ids/OnHurtBeforeMessage.cs
using ModTheSpire.DataDefine;
using ModTheSpire.Unit;

namespace ModTheSpire.Event.Ids
{
    ///受到伤害后
    public class OnHurtBeforeMessage : IEventHandle
    {
        public DamageInfo DamageInfo;
        public AbstractCreature Target;

        public OnHurtBeforeMessage(DamageInfo damageInfo, AbstractCreature target)
        {
            DamageInfo = damageInfo;
            Target = target;
        }
    }
}